solr/core/src/java/org/apache/solr/store/blob/metadata/CorePushPull.java

```diff
@@ -499,7 +500,7 @@ void enqueueForHardDelete(BlobCoreMetadataBuilder bcmBuilder) throws Exception {
     @VisibleForTesting
     protected boolean okForHardDelete(BlobCoreMetadata.BlobFileToDelete file) {
       // For now we only check how long ago the file was marked for delete.
-      return System.nanoTime() - file.getDeletedAt() >= deleteManager.getDeleteDelayMs();
+      return (System.nanoTime() / 1000000) - file.getDeletedAt() >= deleteManager.getDeleteDelayMs();
```

Comment:
From what I could find, deletedAt and deleteDelay is only set using System.nanoTime/1000000, so there should not be any mismatch from comparing times that used nanoTime and times that used currentTimeMillis

Reply:
It might be a good idea to update the BlobFileToDelete class to make it explicit via the javadoc (actually it doesn't have any) that the getDeletedAt() returns the timestamp in milliseconds